docs/static/transforming-data.asciidoc

```diff
@@ -0,0 +1,520 @@
+    code => "
+      map['country_name'] = event.get('country_name')
+      map['towns'] ||= []
+      map['towns'] << {'town_name' => event.get('town_name')}
+      event.cancel()
+    "
+    push_previous_map_as_event => true
+    timeout => 3
+  }
+}
+--------------------------------------------------------------------------------
+
+
+<<plugins-filters-date,date filter>>::
+
+Parses dates from fields to use as Logstash timestamps for events.
++
+The following config parses a field called `logdate` to set the Logstash
+timestamp:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  date {
+    match => [ "logdate", "MMM dd yyyy HH:mm:ss" ]
+  }
+}   
+--------------------------------------------------------------------------------
+
+
+<<plugins-filters-drop,drop filter>>::
+
+Drops events. This filter is typically used in combination with conditionals.
++
+The following config drops 40% of the `debug` level log messages:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  if [loglevel] == "debug" {
+    drop {
+      percentage => 40
+    }
+  }
+}
+--------------------------------------------------------------------------------
+
+
+<<plugins-filters-fingerprint,fingerprint filter>>::
+
+Fingerprints fields by applying a consistent hash.
++
+The following config fingerprints the `IP`, `@timestamp`, and `message` fields
+and adds the hash to a metadata field called `generated_id`:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  fingerprint {
+    source => ["IP", "@timestamp", "message"]
+    method => "SHA1"
+    key => "0123"
+    target => "[@metadata][generated_id]"
+  }
+}
+--------------------------------------------------------------------------------
+
+
+<<plugins-filters-mutate,mutate filter>>::
+
+Performs general mutations on fields. You can rename, remove, replace, and
+modify fields in your events.
++
+The following config renames the `HOSTORIP` field to `client_ip`:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  mutate {
+    rename => { "HOSTORIP" => "client_ip" }
+  }
+}
+--------------------------------------------------------------------------------
+
+
+<<plugins-filters-ruby,ruby filter>>::
+
+Executes Ruby code.
++
+The following config executes Ruby code that cancels 90% of the events:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  ruby {
+    code => "event.cancel if rand <= 0.90"
+  }
+}
+--------------------------------------------------------------------------------
+
+
+[[data-deserialization]]
+=== Deserializing Data
+
+The plugins described in this section are useful for deserializing data into
+Logstash events.
+
+Also see <<filter-plugins>> and <<codec-plugins>> for the full list of available
```

Comment:
Same for this comment as the one above. Maybe just note the full list of filters and codecs are available.